magento-research/pwa-studio
# bug: fix pagination double fetch

## Description

Pagination had some bugs that caused a double fetch. Specifically the `usePagination` hook initialized with a current page of `0`, which is just wrong -- it should default to 1, if anything. So I updated the hook to accept an `initialPage` param which if omitted will default to 1. Now, when we first load, we check the `page` query param and use it if defined.

I ran into another issue, which is that the apollo client will actually just throw the GQL errors, and we never caught them. So, for example, if you request `page=5` and there are only 4 pages, ApolloClient throws an error and we just break. To fix this I added a try-catch and then some retry logic to the category page.

Additional "bonus" fixes:
* Also fixed default page size in gallery items that was causing a visual discrepancy between the "loading" state and the final loaded category page (12 items vs 6 default).
* Also removed some unnecessary cleanup from pagination component and moved it to the category page.

## Related Issue





Closes #1297.

## Verification Steps
1) Load your local and click on a category. It should only fetch the 6 images for the category once.
2) Click on page 2. It should still only fetch the page 2 images only once.
3) CTRL+R to refresh the page. It should still only fetch the images once.
4) In the url, make the query param `page=65` and press enter. It should update to `page=1` and it should only fetch one set of images.

## Screenshots / Screen Captures (if appropriate)

## Proposed Labels for Change Type/Package




`usePagination` now defaults to `1` instead of `0` so this is technically a breaking change.

## Checklist:
